### PR TITLE
Adding compilation flags to control enabling/disabling low priority metrics in concord.

### PR DESCRIPTION
**Pull request by snehal-das**
The compilation flags added below can be turned ON/OFF to enable or disable metrics at compile time.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,6 +47,7 @@ option(BUILD_THIRDPARTY "Wheter to build third party librarie or use preinstalle
 option(CODECOVERAGE "Enable Code Coverage Metrics in Clang" OFF)
 option(ENABLE_RESTART_RECOVERY_TESTS "Enable tests for restart recovery" OFF)
 option(BUILD_UTT "Build UTT library" OFF)
+option(CONCORD_ENABLE_ALL_METRICS "Enable all metrics in concord" ON)
```
Comment (toly-kournik) [inline CMakeLists.txt:50]: How is this option reflects in code?

Comment (snehal-das) [reply CMakeLists.txt:50]: This option has to be removed as it is redundant. It was added when I was exploring ways to add compile time flags. I went with adding the flag in Makefile in this commit.

I need to evaluate if there are other ways to build concord, in which case I **might** need this variable.

**Review by arc-vmware — APPROVED** [review]
LGTM !!!

**Review by cjn — APPROVED** [review]
LGTM. Thanks @snehal-das !

**Comment by snehal-das** [comment]
5 day LRT run was run successfully with these changes added; no issues seen.

**Pull request merged**